### PR TITLE
update release docs, add infos about stable branch

### PR DESCRIPTION
**Pull request by metachris**
## 📝 Summary

Updated release documentation

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,10 +107,14 @@ Ensure you are downloading the most updated MEV-Boost release. Releases are avai
 clone the repository and build it:
 
 ```bash
+# By default, the main branch includes ongoing merged PRs a future release.
 git clone https://github.com/flashbots/mev-boost.git
 cd mev-boost
 
-# If you want to build a specific release, check out the tag first. https://github.com/flashbots/mev-boost/releases
+# You can use the stable branch, which is always updated with the latest released version
+git checkout stable
```
[the remark this answers is not in the document]
Comment (kailinr) [reply README.md:115]: +1 on branch protection.

File: RELEASE.md [hunk 2]
```diff
@@ -14,8 +22,17 @@ make test-race
 go mod tidy
 git status # should be no changes
 
-# Start mev-boost with relay check, and call the mev-boost status endpoint
+# Start mev-boost with relay check and -relays
```
Comment (kailinr) [inline RELEASE.md:25]: I'd prefer a more verbose explanation of what -relay-check and -relays does here (if it is not already explicitly defined). Or, a redirect to check the cli section for more details. *Noticed that -relays is not mentioned in the cli section, by the way. 

"Starts mev-boost on mainnet. `-relay-check `checks relay status on startup and when status API is called. 
.. etc"

OR:

"Check out mev-boost cli arguments for more details"

File: RELEASE.md [hunk 3]
```diff
@@ -61,7 +76,11 @@ To create a new version (with tag), follow all these steps! They are necessary t
...
-* Now push to main and push the tag: `git push && git push --tags`
-* Update `Version` in `config/vars.go` to next patch with `dev` suffix (eg. `v2.3.2-dev`), and commit
+* Update the `stable` branch:
+  * `git checkout stable`
+  * `git merge tags/v2.3.1 --ff-only`
+  * `git checkout main`
+* Update `Version` in `config/vars.go` to next patch with `dev` suffix (eg. `v2.3.2-dev`), commit to main and push to Github
+* Now push the main and stable branch, as well as the tag to Github: `git push origin main stable --tags`
 
 Now check the Github CI actions for release activity: https://github.com/flashbots/mev-boost/actions - CI builds and pushes the Docker image, and prepares a new draft release in https://github.com/flashbots/mev-boost/releases. Open it, generate the description, review, add signoffs and testing, and publish.
```
Comment (kailinr) [inline RELEASE.md:86]: Now check the Github CI actions for release activity: https://github.com/flashbots/mev-boost/actions

- CI builds and pushes the Docker image, and prepares a new draft release in https://github.com/flashbots/mev-boost/releases. 
- Open it, generate the description, review, add sign-offs, tests, and publish.

**Review by kailinr — COMMENTED** [review]
Comments from Sarah and I added

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (metachris) [reply README.md:115]: It has

File: RELEASE.md [hunk 4]
```diff
@@ -61,7 +76,11 @@ To create a new version (with tag), follow all these steps! They are necessary t
...
 * Update `Version` in `config/vars.go` - change it to the next version (eg. from `v2.3.1-dev` to `v2.3.1`), and commit
 * Create a [signed git tag](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-tags): `git tag -s v2.3.1`
-* Now push to main and push the tag: `git push && git push --tags`
-* Update `Version` in `config/vars.go` to next patch with `dev` suffix (eg. `v2.3.2-dev`), and commit
+* Update the `stable` branch:
+  * `git checkout stable`
+  * `git merge tags/v2.3.1 --ff-only`
+  * `git checkout main`
+* Update `Version` in `config/vars.go` to next patch with `dev` suffix (eg. `v2.3.2-dev`), commit to main and push to Github
+* Now push the main and stable branch, as well as the tag to Github: `git push origin main stable --tags`
```
[the remark this answers is not in the document]
Comment (metachris) [reply RELEASE.md:84]: Yes, that's correct and intended

File: RELEASE.md [hunk 5]
```diff
@@ -1,6 +1,14 @@
 # Releasing a new version of mev-boost
 
-This is a guide on how to release a new version of mev-boost.
+This is a guide on how to release a new version of mev-boost:
+
+1. Best days to release a new version are Monday to Wednesday. Never release on a Friday.
```
[the remark this answers is not in the document]
Comment (metachris) [reply RELEASE.md:5]: This renders as sequential numbers!

File: RELEASE.md (hunk 2 above)
Comment (metachris) [reply RELEASE.md:25]: That's details for the readme, not for this release checklist

**Pull request merged**